src/main/java/org/spongepowered/api/context/DisplayNamed.java

```diff
@@ -0,0 +1,33 @@
+package org.spongepowered.api.context;
+
+import org.spongepowered.api.text.Text;
+
+import javax.annotation.Nullable;
+
+/**
+ * Represents an object with a display name.
+ */
+public interface DisplayNamed {
+
+    /**
+     * Gets the current display name for this {@link DisplayNamed}.
+     *
+     * @return The display name
+     */
+    Text getDisplayName();
```

Comment:
Should replicate the other convenience Data API methods by returning `Value<Text>` so that if changed, it can be re-offered back.